pontoon/api/views.py

```diff
@@ -459,3 +464,55 @@ def get_queryset(self):
             raise ValueError(error)
 
         return entities
+
+
+VALID_FORMATS = set(Resource.Format)
+
+
+class PretranslationView(APIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [PersonalAccessTokenAuthentication]
+
+    def post(self, request):
+        resource_format = request.query_params.get("resource_format")
+        locale = request.query_params.get("locale")
+
+        errors = {}
+        try:
+            text = request.body.decode("utf-8")
+        except UnicodeDecodeError:
+            errors["text"] = ["Unable to decode request body as UTF-8."]
+            text = None
+
+        if text is not None:
+            if len(text) > PRETRANSLATION_API_MAX_CHARS:
+                errors["text"] = [
+                    f"Text exceeds maximum length of {PRETRANSLATION_API_MAX_CHARS} characters."
+                ]
+            elif not text.strip():
+                errors["text"] = ["This field is required."]
+
+        if not locale:
+            errors["locale"] = ["This field is required."]
```

Comment:
I wonder if we should also check if `Locale.google_automl_model` is set. That way Pretranslation API will only work for locales for which Pretranslation in Pontoon works, i.e. for the ones that have custom MT models.

@flodolo Thoughts?